### PR TITLE
Fix bug in BoolParam

### PR DESCRIPTION
**Pull request by AVB25**
Fix bug in BoolParam.eval_default() where the default value wasn't properly computed.

### CODE REVIEW

File: test/test_experiment_parameters.py [hunk 1]
```diff
@@ -21,6 +21,21 @@ def test_describe(self):
                 "type": "float"
             })
 
+    def test_evaluate_default(self):
+        def placeholder_get_default(key, default):
+            return key
+
+        param_float_default = IntParam("foo", "bar", 0., min=-1, max=1, unit="baz", scale=1)
```
Comment (dnadlinger) [inline test/test_experiment_parameters.py:28]: This should be `FloatParam`, presumably. I'd also remove all the `min`/`max`/`unit`/`scale` to put the focus on what is acutally being tested here (the default value).

File: test/test_experiment_parameters.py [hunk 2]
```diff
@@ -21,6 +21,21 @@ def test_describe(self):
                 "type": "float"
             })
 
+    def test_evaluate_default(self):
+        def placeholder_get_default(key, default):
```
Comment (dnadlinger) [inline test/test_experiment_parameters.py:25]: As per the parameter name, this function is supposed to be `get_dataset`, not `get_default`. The first parameter is also supposed ot be a string as per the type annotations (although this is not currently enforced at execution time), so the tests are weirdly backwards.

Maybe something like
```py
def mock_get_dataset(key, default=None):
  return {"baz": 42}[key]
```
which would also check that the key is passed correctly, and below `IntParam("foo", "bar", "dataset('baz')")`?

File: test/test_experiment_parameters.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AVB25) [reply test/test_experiment_parameters.py]: Implemented in latest commit

File: test/test_experiment_parameters.py (hunk 2 above)
Comment (AVB25) [reply test/test_experiment_parameters.py]: Other than the previous comment, implemented in previous commit.

Comment (dnadlinger) [reply test/test_experiment_parameters.py]: I was just referring to the dataset() argument order in the previous tests.

**Review by dnadlinger — APPROVED** [review]
Looks good to me – could you please rebase ok top of latest master? The CI checks won't allow this to be merged otherwise.

**Comment by dnadlinger** [comment]
Ah, GitHub now has a button for this – just did it myself.

**Pull request merged**